Sources/TransifexNative/Cache.swift

```diff
@@ -103,73 +119,257 @@ public final class TXDiskCacheProvider: NSObject, TXCacheProvider {
-        self.memoryCache = memoryCache
+                internalCache: TXCache) {
+        super.init(internalCache: internalCache)
         for provider in providers {
             if let providerTranslations = provider.getTranslations() {
-                self.memoryCache.update(translations: providerTranslations,
-                                        replaceEntries: replaceEntries)
+                self.internalCache.update(translations: providerTranslations)
             }
         }
-        super.init()
     }
 }
 
-extension TXProviderBasedCache: TXCache {
-    public func get(key: String, localeCode: String) -> String? {
-        return memoryCache.get(key: key, localeCode: localeCode)
+/// Decorator class responsible for updating the passed internalCache using a certain override policy defined
+/// in the `TXCacheOverridePolicy` enum.
+@objc
+public final class TXStringOverrideFilterCache: TXDecoratorCache {
+    let policy: TXCacheOverridePolicy
+    
+    @objc
+    public init(policy: TXCacheOverridePolicy,
+                internalCache: TXCache) {
+        self.policy = policy
+        super.init(internalCache: internalCache)
     }
     
-    /// Provider based cache doesn't update its internal cache after being initialiazed, so the update
-    /// method of the TXCache protocol is a no-op.
+    override public func update(translations: TXTranslations) {
+        if policy == .overrideAll {
+            super.update(translations: translations)
+            return
+        }
+        
+        var updatedTranslations = self.get()
     
-    public func update(translations: TXTranslations,
-                       replaceEntries: Bool) {
-        /// No-op
+        for (localeCode, localeTranslations) in translations {
+            for (stringKey, translation) in localeTranslations {
+                // Make sure that the new translation has a value and it's not
+                // an empty string.
+                guard let translatedString = translation[TXDecoratorCache.STRING_KEY],
+                      translatedString.count > 0 else {
+                    continue
+                }
+                
+                    // If the policy set set to override untranslated only, then
+                    // update the cache only if there's no existing translation
+                    // for that stringKey.
+                if (policy == .overrideUntranslatedOnly
+                        && self.get(key: stringKey,
+                                localeCode: localeCode) == nil)
+                   ||
+                    // If the policy is set to override using translated only,
+                    // then update the cache only if there's an existing
+                    // translation for that stringKey.
+                    (policy == .overrideUsingTranslatedOnly
+                        && self.get(key: stringKey,
+                                    localeCode: localeCode) != nil) {
```

Comment:
I think this is not correct. `overrideUsingTranslatedOnly` for me means that we'll never replace a non-empty translation in the internal cache with an empty translation found in the incoming translations.

You are already setting `translatedString` with a `guard`, so we know that it's not empty. Therefore, the second conditional should simply be `policy == .overrideUsingTranslatedOnly`.

Reply:
My bad, I thought the `overrideUsingTranslatedOnly` policy meant that the string key should also be found in the cache in order to be updated. I will update the logic there by changing the second conditional as described.

I will also update the docstring for the `overrideUsingTranslatedOnly` value to be:

```
    /// All new translations are added to the cache, either updating existing translations or adding
    /// new ones.
    /// If a translation is not found in cache but exists in the new translations, it's not added.
    /// If a translation is found in cache but doesn't exist in the new translations, it's left untouched.
    /// Empty cache entries from the new translations are filtered out.
```